x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/annotations/Annotation.java

```diff
@@ -54,7 +88,18 @@
         PARSER.declareField(Builder::setModifiedTime,
             p -> TimeUtils.parseTimeField(p, MODIFIED_TIME.getPreferredName()), MODIFIED_TIME, ObjectParser.ValueType.VALUE);
         PARSER.declareString(Builder::setModifiedUsername, MODIFIED_USERNAME);
-        PARSER.declareString(Builder::setType, TYPE);
+        PARSER.declareField(Builder::setType, p -> {
+            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return Type.fromString(p.text());
+            }
+            throw new IllegalArgumentException("Unsupported token [" + p.currentToken() + "]");
+        }, TYPE, ObjectParser.ValueType.STRING);
+        PARSER.declareField(Builder::setEvent, p -> {
+            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return Event.fromString(p.text());
```

Comment:
Since this is generally a lenient parser (`ignoreUnknownFields = true`), I wonder if it should catch a `fromString` exception here and set `event` to `null`.  Otherwise the parser is not lenient in tolerating new `event`s we may add in the future, and it is very likely we will add new `event`s.

Reply:
> Should I make Annotation have two parsers too?

Yes, good point.  Annotation should have a strict parser for parsing C++ output so we don't accidentally do a release where documents the C++ is writing aren't understood by the Java.

The general policy is: strict parsers for user supplied documents and C++ output, lenient parsers for loading documents that might have originated in a different version.

Reply:
I've made the parser strict by:
- setting `ignoreUnknownFields` to `false`
- reverting leniency in enum value parsing

The parser does not seem to be used outside of tests. Once I start using it while reading C++ results, I'll only need strict parser so no need to have both.